pandas/core/indexes/base.py

```diff
@@ -3239,8 +3239,9 @@ def reindex(self, target, method=None, level=None, limit=None,
             if self.equals(target):
                 indexer = None
             else:
-
-                if self.is_unique:
+                # check is_overlapping for IntervalIndex compat
+                if (self.is_unique and
+                        not getattr(self, 'is_overlapping', False)):
```

Comment:
Kind of gross, but I had to do this in a few places since uniqueness isn't sufficient for `IntervalIndex`, as a query on a unique `IntervalIndex` can still have multiple values returned if they come from the overlap of 2+ intervals.

Reply:
why do you need the getattr? isn't self.is_overlapping enough?

Reply:
It's a safeguard since `is_overlapping` is currently only implemented for `IntervalIndex` and this code can hit any type of index.

It could be added to the base `Index` class but it's slightly ambiguous in terms of meaning - it could just simply return `False` or could essentially be `not self.is_unique` (since duplicate points technically overlap).  Would also contribute to API bloat since it'd be public and not really useful outside of `IntervalIndex` for end users. Depends on what tradeoffs we want to make for internal code clarity.